frontend/app/src/components/Sidebar/SidebarNav.test.tsx

```diff
@@ -125,169 +114,144 @@ describe("SidebarNav", () => {
-    const wrapper = shallow(
-      <SidebarNav {...getProps({ hasSidebarElements: true })} />
-    )
+    render(<SidebarNav {...getProps({ hasSidebarElements: true })} />)
 
-    wrapper.find(StyledSidebarNavSeparatorContainer).prop("onClick")!(
-      mockClickEvent
-    )
-    expect(
-      wrapper.find(StyledSidebarNavSeparatorContainer).find(Icon).props()
-    ).toHaveProperty("content", ExpandLess)
+    // Click on the separator to expand the nav component.
+    const separator = screen.getByTestId("stSidebarNavSeparator")
+    expect(separator).toBeInTheDocument()
+    fireEvent.click(separator)
+
+    const collapseIcon = await screen.findByTestId("stSidebarNavCollapseIcon")
+    expect(collapseIcon).toBeInTheDocument()
   })
 
   it("changes cursor to pointer above separator when overflowing", () => {
     mockUseIsOverflowing.mockReturnValueOnce(true)
-    // Need mount > shallow here so that toHaveStyleRule can be used.
-    const wrapper = mount(
-      <SidebarNav {...getProps({ hasSidebarElements: true })} />
-    )
-
-    expect(wrapper.find(StyledSidebarNavSeparatorContainer)).toHaveStyleRule(
-      "cursor",
-      "pointer"
-    )
+    render(<SidebarNav {...getProps({ hasSidebarElements: true })} />)
+    const separator = screen.getByTestId("stSidebarNavSeparator")
+    expect(separator).toBeInTheDocument()
+    expect(separator).toHaveStyleRule("cursor", "pointer")
   })
 
   it("is unexpanded by default", () => {
-    // Need mount > shallow here so that toHaveStyleRule can be used.
-    const wrapper = mount(
-      <SidebarNav {...getProps({ hasSidebarElements: true })} />
-    )
+    render(<SidebarNav {...getProps({ hasSidebarElements: true })} />)
 
-    expect(wrapper.find(StyledSidebarNavItems).prop("isExpanded")).toBe(false)
-    expect(wrapper.find("StyledSidebarNavItems")).toHaveStyleRule(
-      "max-height",
-      "33vh"
-    )
+    const sidebarNavItems = screen.getByTestId("stSidebarNavItems")
+    expect(sidebarNavItems).toBeInTheDocument()
+    expect(sidebarNavItems).toHaveStyleRule("max-height", "33vh")
   })
 
-  it("does not expand when you click on the separator if there is no overflow", () => {
-    const wrapper = shallow(
-      <SidebarNav {...getProps({ hasSidebarElements: true })} />
-    )
+  it("does not expand when you click on the separator if there is no overflow", async () => {
+    render(<SidebarNav {...getProps({ hasSidebarElements: true })} />)
+    const separator = screen.getByTestId("stSidebarNavSeparator")
+    expect(separator).toBeInTheDocument()
+    fireEvent.click(separator)
 
-    wrapper.find(StyledSidebarNavSeparatorContainer).prop("onClick")!(
-      mockClickEvent
-    )
-    expect(wrapper.find(StyledSidebarNavItems).prop("isExpanded")).toBe(false)
+    const sidebarNavItems = await screen.findByTestId("stSidebarNavItems")
+    expect(sidebarNavItems).toBeInTheDocument()
+    expect(sidebarNavItems).toHaveStyleRule("max-height", "33vh")
```

Comment:
Is this necessary? This seems like it would make tests brittle in the future possibly.

Reply:
Sorta necessary since we can't do prop inspection and its the visible consequence of that prop ie: expand/collapse behavior

Reply:
Is finding by testId not enough? Also, isn't the `max-height` just specifying the max-height and not actually if the height is > 0 or another property? 

For example, can't you have `max-height` to be `33vh` but the actual height being `0px` or something?

Reply:
Testid just tells us it rendered, but not that the click was handled the way we expect. 

Maybe I'm confused, but since the original test inspected that the prop `isExpanded` was false, needed some parallel test to ensure expected behavior. 
When `isExpanded` prop is false, the maxHeight is set to "33vh" instead of "75vh" so its not necessarily about testing height, its testing the `isExpanded` by proxy.